### PR TITLE
Adding pagination to disputes grid

### PR DESCRIPTION
**Pull request by eburgos**
Refers to #62

Disputes grid with pagination. I hope it helps.

Regards,

### CODE REVIEW

File: src/Components/DisputesTable/GridContent/index.js [hunk 1]
```diff
@@ -8,13 +8,35 @@ import { truncateText } from '../../../helpers/truncateText'
...
 
-class GridContent extends Component {
+export class GridContent extends Component {
+  constructor () {
+    super()
+    this.state = {
+      currentPage: 1
+    }
+    this.paginateHandler = this.paginateHandler.bind(this)
+    this.paginateInputHandler = this.paginateInputHandler.bind(this)
```
Comment (n1c01a5) [inline src/Components/DisputesTable/GridContent/index.js:18]: I think you don't need `bind()` see https://medium.com/shoutem/react-to-bind-or-not-to-bind-7bf58327e22a

File: src/Components/DisputesTable/GridContent/index.js [hunk 2]
```diff
@@ -8,13 +8,35 @@ import { truncateText } from '../../../helpers/truncateText'
...
+  }
   componentWillMount () {
     this.props.getDataDisputes()
   }
 
+  paginateHandler (event) {
```
Comment (n1c01a5) [inline src/Components/DisputesTable/GridContent/index.js]: could use the functional syntax for your own function like `paginateHandler = e => (this.setState({ currentPage: Number(e.target.dataset.pageNumber) }))`

File: src/Components/DisputesTable/GridContent/index.js [hunk 3]
```diff
@@ -8,13 +8,35 @@ import { truncateText } from '../../../helpers/truncateText'
...
+    this.setState({
+      currentPage: Number(event.target.dataset.pageNumber)
+    })
+  }
+  paginateInputHandler (event) {
+    this.setState({
+      currentPage: Number(event.target.value)
+    })
+  }
```
Comment (n1c01a5) [inline src/Components/DisputesTable/GridContent/index.js:33]: idems

File: src/Components/DisputesTable/GridContent/index.js [hunk 4]
```diff
@@ -32,6 +54,14 @@ class GridContent extends Component {
...
 
     const filteredDisputes = filterFunction(disputes)
+    const indexOfLastRow = currentPage * rowsPerPage
+    const indexOfFirstRow = indexOfLastRow - rowsPerPage
+    const list = filteredDisputes.filter(dispute => {
+      return dispute.arbitratorAddress === process.env.REACT_APP_ARBITRATOR_ADDRESS
+    })
+    const currentRows = list.slice(indexOfFirstRow, indexOfLastRow)
+
+    const totalPages = Math.ceil(list.length / rowsPerPage) || 1
```
Comment (n1c01a5) [inline src/Components/DisputesTable/GridContent/index.js:64]: can you create a function like `const currentRows = (list, indexOfLastRow, indexOfFirstRow)` in the `utils` folder

**Review by n1c01a5 — CHANGES_REQUESTED** [review]

**Comment by n1c01a5** [comment]
and can you hide the "paginator" if there are not enough disputes

File: src/Components/DisputesTable/GridContent/index.js (hunk 1 above)
Comment (eburgos) [reply src/Components/DisputesTable/GridContent/index.js:18]: Inside those method I'm settings this.state. If I don't bind then I won't have this available (would be undefined). Please let me know if you consider a workaround.

File: src/Components/DisputesTable/GridContent/index.js (hunk 4 above)
Comment (eburgos) [reply src/Components/DisputesTable/GridContent/index.js:64]: I couldn't find a utils folder. Perhaps you meant "helpers" ?

**Comment by eburgos** [comment]
@n1c01a5 Fixed most of the stuff you requested. 
1) I couldn't find that "utils" folder, perhaps you meant the "helpers" one
2) Regarding the .bind(this) I need to use "this" inside the function so I can't skip that bind. Unless there is another way of achieving this that I'm not aware.

**Comment by n1c01a5** [comment]
@eburgos yes it's the `helpers` folder and yes I think we can skip it but it's about the refactoring. We can see that later. Tnx for your contribution.

**Pull request merged**